src/Extension.ml

```diff
@@ -11,37 +11,39 @@ let openTerminalSelectCommandId = "ocaml.open-terminal-select"
 let switchImplIntfCommandId = "ocaml.switch-impl-intf"
 
 module Client = struct
-  let make () : Vscode.LanguageClient.clientOptions =
-    let documentSelector : Vscode.LanguageClient.documentSelectorItem array =
-      [| { scheme = "file"; language = "ocaml" }
-       ; { scheme = "file"; language = "ocaml.interface" }
-       ; { scheme = "file"; language = "ocaml.ocamllex" }
-       ; { scheme = "file"; language = "ocaml.menhir" }
-       ; { scheme = "file"; language = "reason" }
-      |]
+  let make () : LanguageClient.ClientOptions.t =
+    let document_selector =
+      LanguageClient.DocumentSelector.
+        [| language "ocaml"
+         ; language "ocaml.interface"
+         ; language "ocaml.ocamllex"
+         ; language "ocaml.menhir"
+         ; language "reason"
+        |]
     in
-    let (lazy outputChannel) = Output.languageServerOutputChannel in
-    let revealOutputChannelOn =
-      Vscode.LanguageClient.RevealOutputChannelOn.tToJs Never
-    in
-    Vscode.LanguageClient.clientOptions ~documentSelector ~outputChannel
-      ~revealOutputChannelOn ()
+    let (lazy output_channel) = Output.languageServerOutputChannel in
+    let reveal_output_channel_on = LanguageClient.RevealOutputChannelOn.Never in
+    LanguageClient.ClientOptions.create ~document_selector ~output_channel
+      ~reveal_output_channel_on ()
 end
 
 module Server = struct
-  let make (toolchain : Toolchain.resources) :
-      Vscode.LanguageClient.serverOptions =
+  let make (toolchain : Toolchain.resources) : LanguageClient.ServerOptions.t =
     let command = Toolchain.getLspCommand toolchain in
     Cmd.log command;
-    let env = Process.env in
+    let env = Process.env () in
     match command with
-    | Shell commandLine ->
-      { command = commandLine; args = [||]; options = { env; shell = true } }
+    | Shell command ->
+      let options =
+        LanguageClient.ExecutableOptions.create ~env ~shell:true ()
+      in
+      LanguageClient.Executable.create ~command ~options ()
     | Spawn { bin; args } ->
-      { command = Path.toString bin
-      ; args = Array.of_list args
-      ; options = { env; shell = false }
-      }
+      let command = Path.toString bin in
+      let options =
+        LanguageClient.ExecutableOptions.create ~env ~shell:false ()
+      in
+      LanguageClient.Executable.create ~command ~args ~options ()
 end
```

Comment:
What do you think if we replace these single function modules with just functions? `createServerOptions ()` or `serverOptions ()`